### PR TITLE
Add CleanGeoStreamR package

### PR DESCRIPTION
**Pull request by ilhan-mutlu**
Describe your pull request here

Please add the newly developed CleanGeoStreamR package.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/cleangeostreamr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  git_url: https://codebase.helmholtz.cloud/department-computational-biology/software/cleangeostreamr.git
+  git_tag: v{{ version }}
+
+build:
+  noarch: generic
```
Comment (bernt-matthias) [inline recipes/cleangeostreamr/meta.yaml:13]:
```suggestion
  build: 0
  noarch: generic
```

File: recipes/cleangeostreamr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
+{% set name = "cleangeostreamr" %}
+{% set version = "1.0.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  git_url: https://codebase.helmholtz.cloud/department-computational-biology/software/cleangeostreamr.git
+  git_tag: v{{ version }}
```
Comment (bernt-matthias) [inline recipes/cleangeostreamr/meta.yaml:10]:
```suggestion
  url: "https://codebase.helmholtz.cloud/department-computational-biology/software/{{ name }}/-/archive/v{{ version }}/{{ name }}-v{{ version }}.tar.gz"
  sha256: 6de64677932558cd1157a8dc11f1115db3fe5eeb0e628fbe7111564d9aecc6fc
```

File: recipes/cleangeostreamr/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    - r-jsonlite
+    - r-magrittr
+    - r-geosphere
+    - r-stringi
+    - r-dplyr
+    - r-ggplot2
+    - r-stringr
+    - r-tibble
+    - r-tidyr
+
```
Comment (bernt-matthias) [inline recipes/cleangeostreamr/meta.yaml:51]:
```suggestion

test:
  commands:
    - $R -e "library('cleangeostreamr')"
```

File: recipes/cleangeostreamr/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+{% set name = "cleangeostreamr" %}
+{% set version = "1.0.0" %}
+
+package:
+  name: {{ name }}
```
Comment (bernt-matthias) [inline recipes/cleangeostreamr/meta.yaml:5]: Most conda R packages have a prefix `r-`

```suggestion
  name: r-{{ name }}
```

**Review by bernt-matthias — COMMENTED** [review]
Few suggestion. 

You also need to pack the GPL license file here.

File: recipes/cleangeostreamr/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+source:
+  git_url: https://codebase.helmholtz.cloud/department-computational-biology/software/cleangeostreamr.git
+  git_tag: v{{ version }}
+
+build:
+  noarch: generic
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  script: R CMD INSTALL --build .
```
Comment (bernt-matthias) [inline recipes/cleangeostreamr/meta.yaml:17]: Should all R scripts be executable? Then 

```suggestion
  script: R CMD INSTALL --build . && chmod +x R/*.R && cp R/*.R ${PREFIX}/bin/
```

If it's not all `*.R` scripts than you may copy and chmod them individually. Maybe its also better to have it in a separate `build.sh` like here: https://github.com/bioconda/bioconda-recipes/blob/0db5639941324f8b3dc21fceb084f1d8284e8286/recipes/bioconductor-maaslin2/build.sh

File: recipes/cleangeostreamr/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  url: "https://codebase.helmholtz.cloud/department-computational-biology/software/{{ name }}/-/archive/v{{ version }}/{{ name }}-v{{ version }}.tar.gz"
+  sha256: 6de64677932558cd1157a8dc11f1115db3fe5eeb0e628fbe7111564d9aecc6fc
+
+build:
+  build: 0
+  noarch: generic
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  script: R CMD INSTALL --build .
```
Comment (bernt-matthias) [inline recipes/cleangeostreamr/meta.yaml:18]:
```suggestion
  script: R CMD INSTALL --build .
  run_exports:
    - {{ pin_subpackage("r-cleangeostreamr", max_pin="x.x") }}
```

File: recipes/cleangeostreamr/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+package:
+  name: r-{{ name }}
+  version: {{ version }}
+
+source:
+  url: "https://codebase.helmholtz.cloud/department-computational-biology/software/{{ name }}/-/archive/v{{ version }}/{{ name }}-v{{ version }}.tar.gz"
+  sha256: 6de64677932558cd1157a8dc11f1115db3fe5eeb0e628fbe7111564d9aecc6fc
+
+build:
+  build: 0
```
Comment (bernt-matthias) [inline recipes/cleangeostreamr/meta.yaml:13]: My bad:

```suggestion
  number: 0
```

File: recipes/cleangeostreamr/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,68 @@
+{% set name = "cleangeostreamr" %}
+{% set version = "1.0.0" %}
+
+package:
+  name: r-{{ name }}
```
Comment (bernt-matthias) [inline recipes/cleangeostreamr/meta.yaml:5]: We will also need to rename the directory to `r-cleangeostreamr`.

**Comment by bernt-matthias** [comment]
> In the "about" part license info was included as "license: GPL-3.0". Does this sufficient or the main license file in the original repository of the CleanGeoStreamR should be linked?

Can you try to add `license_file: LICENSE.md`

**Comment by ilhan-mutlu** [comment]
> > In the "about" part license info was included as "license: GPL-3.0". Does this sufficient or the main license file in the original repository of the CleanGeoStreamR should be linked?
> 
> Can you try to add `license_file: LICENSE.md`

I think I upload the LICENSE.md file now.

**Comment by bernt-matthias** [comment]
Tests fail: 

```
16:52:19 BIOCONDA INFO (OUT)   - nothing provides requested r-grdevices
16:52:19 BIOCONDA INFO (OUT)   - nothing provides requested r-parallel
```

I guess we won't need and grdevices explicitly (should come if needed implicitly).
Also parallel is probably not needed, see also https://codebase.helmholtz.cloud/department-computational-biology/software/cleangeostreamr/-/issues/4

**Comment by bernt-matthias** [comment]
We have the following error: `Empty package; r-base present in build and host deps. `. I think you can remove the complete `build` section. See for instance here: https://github.com/bernt-matthias/bioconda-recipes/blob/0db5639941324f8b3dc21fceb084f1d8284e8286/recipes/bioconductor-maaslin2/meta.yaml

File: recipes/r-cleangeostreamr/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+  noarch: generic
+  rpaths:
+    - lib/R/lib/
+    - lib/
+  run_exports:
+    - {{ pin_subpackage("r-cleangeostreamr", max_pin="x.x") }}
+
+requirements:
+  build:
+    - r-base
```
Comment (bernt-matthias) [inline recipes/r-cleangeostreamr/meta.yaml:23]:
```suggestion
```

**Comment by bernt-matthias** [comment]
The test fails: 

```
20:32:07 BIOCONDA INFO (OUT) Error in library("cleangeostreamr") :
20:32:07 BIOCONDA INFO (OUT)   there is no package called ‘cleangeostreamr’
```

Could it be that capitalisation matters: i.e. you should `CleanGeoStreamR`. In this case I would suggest to redefine it in the package to all lower case... or?

File: recipes/r-cleangeostreamr/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+    - r-stringi
+    - r-dplyr
+    - r-ggplot2
+    - r-stringr
+    - r-tibble
+    - r-tidyr
+
+test:
+  commands:
+    - $R -e "library('cleangeostreamr')"
```
Comment (bernt-matthias) [inline recipes/r-cleangeostreamr/meta.yaml:71]:
```suggestion
    - $R -e "library('CleanGeoStreamR')"
```

**Comment by ilhan-mutlu** [comment]
> The test fails:
> 
> ```
> 20:32:07 BIOCONDA INFO (OUT) Error in library("cleangeostreamr") :
> 20:32:07 BIOCONDA INFO (OUT)   there is no package called ‘cleangeostreamr’
> ```
> 
> Could it be that capitalisation matters: i.e. you should `CleanGeoStreamR`. In this case I would suggest to redefine it in the package to all lower case... or?

That could be root cause. I saw that you already committed modifications. Let us wait for the test results.

File: recipes/r-cleangeostreamr/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+    - r-stringi
+    - r-dplyr
+    - r-ggplot2
+    - r-stringr
+    - r-tibble
+    - r-tidyr
+
+test:
+  commands:
+    - $R -e "library('CleanGeoStreamR')"
```
Comment (bernt-matthias) [inline recipes/r-cleangeostreamr/meta.yaml:73]: Last but not least can you add a test for the executable scripts, e.g. using the `--help` argument?

**Pull request merged**